### PR TITLE
Ship scipp with conan cmake setup to simplify dependency install

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Main beneficiaries: `cmake-package-test` (called by `conda build` in release workflow) and `scippneutron`.

### CODE REVIEW

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -298,9 +224,10 @@ if(DYNAMIC_LIB)
   install(
     FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake"
           "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config-version.cmake"
+          "${CMAKE_CURRENT_SOURCE_DIR}/cmake/${PROJECT_NAME}-conan.cmake"
     DESTINATION lib/cmake/scipp
```
Comment (jl-wynen) [inline lib/CMakeLists.txt:228]: Should this also use `PROJECT_NAME`? (Same below)

File: lib/cmake/scipp-config.cmake.in [hunk 2]
```diff
@@ -4,12 +4,25 @@
...
-check_required_components(scipp-targets)
+  check_required_components(scipp-targets)
 
-include(CMakeFindDependencyMacro)
-find_dependency(Boost 1.69)
-find_dependency(Eigen3)
-find_dependency(LLNL-Units)
-find_dependency(TBB)
+  include(CMakeFindDependencyMacro)
+  find_dependency(Boost 1.69)
```
Comment (jl-wynen) [inline lib/cmake/scipp-config.cmake.in:24]: This version does not match the one in `scipp-conan.cmake`. Does it have to be specified here?

**Review by jl-wynen — COMMENTED** [review]
Please update the documentation on locations of dependencies.

File: lib/CMakeLists.txt (hunk 1 above)
Comment (SimonHeybrock) [reply lib/CMakeLists.txt:228]: It is? What are you referring to?

Comment (SimonHeybrock) [reply lib/CMakeLists.txt:228]: Oh, you don't mean the changed line, but the line below?

File: lib/cmake/scipp-config.cmake.in (hunk 2 above)
Comment (SimonHeybrock) [reply lib/cmake/scipp-config.cmake.in:24]: This here defines the minimum required version. `scipp-conan.cmake` defines the one you get when you use conan to install boost. If someone wants to use scipp without conan they still need 1.69 or later, so it has to be specified here.

File: lib/CMakeLists.txt (hunk 1 above)
Comment (jl-wynen) [reply lib/CMakeLists.txt]: Yes

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**